.github/workflows/queasars_deployment.yml

```diff
@@ -0,0 +1,94 @@
+# for consistency with the pyproject.toml version and then deploys to PYPI and GitHub pages.
+
+name: QUEASARS Deployment
+
+on:
+  # Runs only on protected tags of the form vX.Y.Z
+  push:
+    tags: ['v[0-9].[0-9].[0-9]']
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version of the release in the form: vX.Y.Z'
+        type: string
+        required: true
+      deploy_to_pypi:
+        description: 'Boolean value indicating whether QUEASARS should be deployed to PyPI.'
+        type: boolean
+        required: false
+        default: true
+      deploy_to_pages:
+        description: 'Boolean value indicating whether QUEASARS documentation should be deployed to GitHub Pages.'
+        type: boolean
+        required: false
+        default: true
+
+concurrency:
+  # Only let one deployment run at a time.
+  group: "queasars deployment"
+  cancel-in-progress: false
+
+jobs:
+  Check_Version_Consistency:
+    # Check that the project version given by the tag / manual workflow input matches the pyproject.toml version
+    name: Check Version Consistency
+    runs-on: ubuntu-latest
+    steps:
+      - if: ${{ inputs.version != '' }}
+        name: Check workflow version input
+        shell: bash
+        run: |
+          if [[ $inputs.version != @(v[0-9].[0-9].[0-9]) ]]; then exit 1; fi
+          echo "version=${inputs.version:1}" >> $GITHUB_ENV
+      - if: ${{ inputs.version == '' }}
+        name: Get tag version
+        shell: bash
+        env:
+          TAG_NAME: ${{ github.ref_name }}
+        run: |
+          echo "version=${TAG_NAME:1}" >> $GITHUB_ENV
+      - uses: actions/checkout@v3
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v3
+        with:
+          python-version: "3.11"
+      - name: Install pipx
+        run: |
+          python3 -m pip install --upgrade pip
+          python3 -m pip install --user pipx
+          python3 -m pipx ensurepath
+      - name: Install poetry
+        run: |
+          pipx install poetry==1.7.0
+      - name: Install dependencies
+        run: |
+          poetry install
+      - name: Check tag / input version consistency with pyproject.toml
+        shell: bash
+        run: |
+          echo "project version derived from tag / manual input: ${version}"
+          toml_version=$(poetry version --short)
+          echo "project version from pyproject.toml ${toml_version}"
+          if [[ $version != $toml_version ]]; then exit 1; fi
+
+  Run_CI:
+    # Run the CI again before deploying.
+    name: Run CI
+    uses: ./.github/workflows/queasars_ci.yml
+
+  Deploy_To_PyPI:
+    # Only deploy to PyPI if both the project version is consistent and the CI is run successfully.
+    name: Deploy to PyPI
+    needs: [Check_Version_Consistency, Run_CI]
+    if: ${{ inputs.deploy_to_pypi == '' || inputs.deploy_to_pypi == true }}
```

Comment:
What is this condition for?

Reply:
The workflow can be dispatched manually from the github actions tab. There you can specify as an input, whether you want to deploy to just pypi, to just github pages or both. This might be useful if the deployment to pypi succeeded but the one to github pages failed and you want to manually rerun the documentation deployment. The condition here checks whether the deployment to pypi was chosen in the manual dispatch, or  whether the workflow was triggered from the tag and therefore no such input exists.

Reply:
Copy, understood!